### PR TITLE
New & improved Pike syntax colouring.

### PR DESCRIPTION
**Pull request by BuGlessRB**
This is a long overdue update of the Pike language syntax colouring support.

It now supports Pike 8.1, embedded SQL, embedded Autodoc and numerous other fixes.

### CODE REVIEW

**Comment by BuGlessRB** [comment]
I presume that I can ignore the two automated and failed coverage/coveralls and continuous-integration/travis-ci/pr checks?

File: runtime/syntax/pike.vim [hunk 1]
```diff
@@ -1,142 +1,385 @@
 " Vim syntax file
 " Language:	Pike
-" Maintainer:	Francesco Chemolli <kinkie@kame.usr.dsi.unimi.it>
```
Comment (k-takata) [inline runtime/syntax/pike.vim]: Could you change this line to "Previous Maintainer:" instead of removing?

Comment (BuGlessRB) [reply runtime/syntax/pike.vim]: Fixed.

**Comment by BuGlessRB** [comment]
I folded all commits into a single commit for easy reviewing.

**Comment by BuGlessRB** [comment]
There's two commits now:
1. Updates pike.vim, adds support for cmod.vim and autodoc.vim.
2. Updates c.vim to add optional support for embedded Autodoc.

**Comment by BuGlessRB** [comment]
For reviewing use the following link: https://github.com/vim/vim/pull/2558
Due to the rebase/forced push, the older links might not work anymore.

**Comment by BuGlessRB** [comment]
Adjusted the unit tests for the mapping of lpc/ulpc and cmod filetypes.

**Comment by BuGlessRB** [comment]
Is there anything I can do to expedite the review and get these changes committed to the main repository?

**Comment by brammool** [comment]
I'll include it, thanks.
It would be nice to document the variables in the  help.

**Pull request closed without merging**